packages/components/addon/components/hds/form/character-count/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+export const ID_PREFIX = 'character-count-';
+
+const NOOP = () => {};
+
+export default class HdsFormCharacterCountIndexComponent extends Component {
+  @tracked message;
+  @tracked currentLength; // The current number of characters typed into the associated input
+
+  /**
+   * @param maxLength
+   * @type {number}
+   * @default null
+   * @description The maximum number of characters allowed.
+   */
+  get maxLength() {
+    let { maxLength } = this.args;
+    return maxLength;
+  }
+
+  /**
+   * @param remainingLength
+   * @type {number}
+   * @default null
+   * @description The remaining number of characters.
+   */
+  get remainingLength() {
+    return this.maxLength - this.currentLength;
+  }
+
+  /**
+   * Determines the unique ID to assign to the element
+   * @method id
+   * @return {(string|null)} The "id" attribute to apply to the element or null, if no controlId is provided
+   */
+  get id() {
+    let { controlId } = this.args;
+    if (controlId) {
+      return `${ID_PREFIX}${controlId}`;
+    }
+    return null;
+  }
+
+  /**
+   * @param onInsert
+   * @type {function}
+   * @default () => {}
+   */
+  get onInsert() {
+    let { onInsert } = this.args;
+
+    let inputControl = document.getElementById(this.args.controlId);
+    if (inputControl && inputControl.value) {
+      this.updateMessage(inputControl);
+      inputControl.addEventListener('input', this.onInput);
```

Comment:
we never remove this event listener. can you think of a way of doing it?

Reply:
The removal of this event listener is handled by the garbage collector in most cases (except for when the CharacterCount is removed by the associated input remains on the page). Happy to add the logic and deal with this  https://github.com/hashicorp/design-system/pull/1802/commits/8523c3f27e700455d542dfeb976044d32f37441d